### DISCUSSION TITLE
Key Pair Generation and Storage in EJBCA

### DISCUSSION THREAD

**Discussion by hzami** [Keyfactor/ejbca-ce#608] [Q&A]
Hi !
 I have a problem understanding the process of requesting a certificate with key pairs generated by the CA. Does this mean that the key pair is generated and then stored in the database so the CA can send it later to the user (requester)?
If yes, where is it stored (  the private key) ? In which table?
If not, how can the user get their private key?

**Comment by primetomas** [comment]
End entity key pairs are never stored by EJBCA, unless key recovery is enabled, which it is not by default.
https://docs.keyfactor.com/ejbca/latest/key-recovery

If you make a request for server generated keys (without key recovery), the key pair is generated in memory, and discarded once sent back to the client. The client receives the key and certificate, for example in a PKCS#12 keystore, and after that it's the clients responsibility to keep track of it.
